src/main/java/core/basesyntax/service/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.FruitsForSale;
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void updateDataBase(FruitTransaction transaction) {
+        FruitsForSale fruitsForSale = new FruitsForSale();
+        if (fruitsForSale.isNotForSale(transaction.getFruit())) {
+            throw new RuntimeException("We don't sell such fruit: " + transaction.getFruit());
+        }
+        if (FruitStorage.fruitStorage.get(transaction.getFruit()) == null) {
+            throw new RuntimeException("First you need to enter the balance of "
+                    + transaction.getFruit() + " under the code 'b'");
+        }
+        for (Map.Entry<String, Integer> entry : FruitStorage.fruitStorage.entrySet()) {
+            if (transaction.getFruit().equals(entry.getKey())) {
+                if (entry.getValue() - transaction.getQuantity() < 0) {
+                    throw new RuntimeException("The balance after the transaction"
+                            + " cannot be negative.");
```

Comment:
The error message could be more informative by including the fruit name and the attempted purchase quantity. This would help in debugging and understanding which transaction caused the error.